<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SalesmanMainWidget</class>
 <widget class="QMainWindow" name="SalesmanMainWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>910</width>
    <height>516</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Эволюционные вычисления</string>
  </property>
  <property name="windowIcon">
   <iconset resource="images.qrc">
    <normaloff>:/img/img/evoluting150px.png</normaloff>:/img/img/evoluting150px.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <widget class="QLabel" name="labelSizeMatrixWeight">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string>Размер матрицы весов</string>
      </property>
      <property name="buddy">
       <cstring>sizeMatrixWeight</cstring>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QSpinBox" name="sizeMatrixWeight">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="suffix">
       <string/>
      </property>
      <property name="minimum">
       <number>2</number>
      </property>
      <property name="maximum">
       <number>10</number>
      </property>
      <property name="value">
       <number>5</number>
      </property>
     </widget>
    </item>
    <item row="0" column="2" colspan="2">
     <spacer name="horizontalSpacer">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>558</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="1" column="0" colspan="3">
     <widget class="QTableView" name="view">
      <property name="dragDropMode">
       <enum>QAbstractItemView::DropOnly</enum>
      </property>
      <property name="selectionMode">
       <enum>QAbstractItemView::NoSelection</enum>
      </property>
      <property name="textElideMode">
       <enum>Qt::ElideMiddle</enum>
      </property>
     </widget>
    </item>
    <item row="1" column="3" rowspan="3">
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="1">
       <widget class="QLCDNumber" name="lcdNumberSizePopulation">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Maximum" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>30</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="1" column="0" colspan="2">
       <widget class="QSlider" name="sizePopulation">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimum">
         <number>2</number>
        </property>
        <property name="maximum">
         <number>50</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="tickPosition">
         <enum>QSlider::TicksBothSides</enum>
        </property>
        <property name="tickInterval">
         <number>5</number>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLCDNumber" name="lcdNumberCountReproduction">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Maximum" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>30</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="11" column="0" colspan="2">
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="4" column="1">
       <widget class="QLCDNumber" name="lcdNumberCountMutation">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Maximum" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>30</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="labelCountReproduction">
        <property name="text">
         <string>Количество скрешеваний</string>
        </property>
        <property name="buddy">
         <cstring>countReproduction</cstring>
        </property>
       </widget>
      </item>
      <item row="3" column="0" colspan="2">
       <widget class="QSlider" name="countReproduction">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="tickPosition">
         <enum>QSlider::TicksBothSides</enum>
        </property>
        <property name="tickInterval">
         <number>5</number>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="labelCountMutation">
        <property name="text">
         <string>Количество мутаций</string>
        </property>
        <property name="buddy">
         <cstring>countMutation</cstring>
        </property>
       </widget>
      </item>
      <item row="5" column="0" colspan="2">
       <widget class="QSlider" name="countMutation">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="tickPosition">
         <enum>QSlider::TicksBothSides</enum>
        </property>
        <property name="tickInterval">
         <number>5</number>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="labelSizePopulation">
        <property name="text">
         <string>Размер популяции</string>
        </property>
        <property name="buddy">
         <cstring>sizePopulation</cstring>
        </property>
       </widget>
      </item>
      <item row="10" column="0" colspan="2">
       <widget class="QPushButton" name="beginCalculation">
        <property name="text">
         <string>Начать вычисления</string>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="QLCDNumber" name="lcdNumberCountIterationCalc">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Maximum" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>30</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="labelCountIterationCalc">
        <property name="text">
         <string>Количество итераций вычислений</string>
        </property>
        <property name="buddy">
         <cstring>countMutation</cstring>
        </property>
       </widget>
      </item>
      <item row="8" column="0" colspan="2">
       <widget class="QSlider" name="countIterationCalc">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>100</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="tickPosition">
         <enum>QSlider::TicksBothSides</enum>
        </property>
        <property name="tickInterval">
         <number>5</number>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="2" column="0">
     <widget class="QLabel" name="labelCalculation">
      <property name="text">
       <string>Лог вычислений</string>
      </property>
      <property name="buddy">
       <cstring>logCalculation</cstring>
      </property>
     </widget>
    </item>
    <item row="2" column="2">
     <widget class="QProgressBar" name="progressBarCalculation">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="value">
       <number>0</number>
      </property>
      <property name="textVisible">
       <bool>false</bool>
      </property>
     </widget>
    </item>
    <item row="3" column="0" colspan="3">
     <widget class="QPlainTextEdit" name="logCalculation">
      <property name="documentTitle">
       <string>Результат вычислений</string>
      </property>
      <property name="undoRedoEnabled">
       <bool>false</bool>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>910</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>&amp;Файл</string>
    </property>
    <addaction name="paste"/>
    <addaction name="separator"/>
    <addaction name="quite"/>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>&amp;Справка</string>
    </property>
    <addaction name="about"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_2"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="paste">
   <property name="text">
    <string>Вставить таблицу из буфера обмена</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="quite">
   <property name="text">
    <string>Выход</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="about">
   <property name="text">
    <string>О программе</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>sizePopulation</tabstop>
  <tabstop>countReproduction</tabstop>
  <tabstop>countMutation</tabstop>
 </tabstops>
 <resources>
  <include location="images.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>sizePopulation</sender>
   <signal>valueChanged(int)</signal>
   <receiver>lcdNumberSizePopulation</receiver>
   <slot>display(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>811</x>
     <y>121</y>
    </hint>
    <hint type="destinationlabel">
     <x>899</x>
     <y>86</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>countReproduction</sender>
   <signal>valueChanged(int)</signal>
   <receiver>lcdNumberCountReproduction</receiver>
   <slot>display(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>899</x>
     <y>192</y>
    </hint>
    <hint type="destinationlabel">
     <x>899</x>
     <y>157</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>countMutation</sender>
   <signal>valueChanged(int)</signal>
   <receiver>lcdNumberCountMutation</receiver>
   <slot>display(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>899</x>
     <y>263</y>
    </hint>
    <hint type="destinationlabel">
     <x>899</x>
     <y>228</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>countIterationCalc</sender>
   <signal>valueChanged(int)</signal>
   <receiver>lcdNumberCountIterationCalc</receiver>
   <slot>display(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>767</x>
     <y>323</y>
    </hint>
    <hint type="destinationlabel">
     <x>867</x>
     <y>282</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>beginCalculation</sender>
   <signal>clicked()</signal>
   <receiver>logCalculation</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>726</x>
     <y>353</y>
    </hint>
    <hint type="destinationlabel">
     <x>558</x>
     <y>354</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>paste</sender>
   <signal>triggered()</signal>
   <receiver>SalesmanMainWidget</receiver>
   <slot>paste()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>454</x>
     <y>257</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>quite</sender>
   <signal>triggered()</signal>
   <receiver>SalesmanMainWidget</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>454</x>
     <y>257</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>about</sender>
   <signal>triggered()</signal>
   <receiver>SalesmanMainWidget</receiver>
   <slot>aboutProgram()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>454</x>
     <y>257</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>paste()</slot>
  <slot>aboutProgram()</slot>
 </slots>
</ui>
